web/src/invite.ts

```diff
@@ -117,6 +118,29 @@ function get_common_invitation_data(): CommonInvitationData {
             data.invitee_emails += "," + current_email;
         }
     }
+
+    if (current_user.is_admin) {
+        const realm_welcome_message_configured = realm.realm_welcome_message_custom_text.length > 0;
+        const send_realm_default_custom_message = $(
+            "#send_default_realm_welcome_message_custom_text",
+        ).is(":checked");
+        const send_custom_message = $("#send_custom_welcome_message_custom_text").is(":checked");
+        const welcome_message_custom_text = $<HTMLTextAreaElement>(
+            "#invite_welcome_custom_message_text",
+        )
+            .val()!
+            .trim();
+
+        if (
+            (realm_welcome_message_configured && !send_realm_default_custom_message) ||
+            (!realm_welcome_message_configured &&
```

Comment:
This becomes quite difficult to read. I think you can form a function like `get_welcome_message_custom_text_value` and then having a separate block for each condition would be better -
```js
function get_welcome_message_custom_text_value() {
    if (realm_welcome_message_configured) {
        if (send_realm_default_custom_message) {
            data.welcome_message_custom_text = JSON.stringify(null)
        } else {
            data.welcome_message_custom_text = welcome_message_custom_text;
        }
        return;
    }
.....
}
```